### DISCUSSION TITLE
How do I plot in a specific location defined by a Rect ?

### DISCUSSION THREAD

**Discussion by vortex314** [emilk/egui#3969] [Q&A] [answered]
The plot interface doesn´t behave like any other widget. So it looks that I cannot use ui.put(rect,widget).
What is the way to do this ?
The purpose is to show a realtime graph of incoming data. 
Iḿ working on a configuration driven dashboard for MQTT and Redis PubSub : https://github.com/vortex314/egui-dashboard
Any help is welcome ! 
All the examples I found doesn´t seem to work this way.

**Comment by emilk** [comment, marked as answer]
You can use [`ui.child_ui`](https://docs.rs/egui/latest/egui/struct.Ui.html#method.child_ui) to positions a `Ui` anywhere you want, and then put the plot in that new ui

**Comment by vortex314** [comment]
Works as expected. Thanks a lot ! @emilk 
Actual code 
```rust
        let line = egui_plot::Line::new(line_points);
        let pl = egui_plot::Plot::new("example_plot")
            .height(rect.height())
            .width(rect.width())
            .show_axes(true)
            .show_grid(true)
            .data_aspect(1.0);
        let layout = Layout::top_down(Align::LEFT);
        info!("Plot {} : {:?}", self.label, self.rect);
        let mut child_ui = ui.child_ui(self.rect, layout);
        let _r  = pl.show(&mut child_ui, |plot_ui| {
            plot_ui.line(line);
        })
        .response;
```
